public/src/js/models/article/editor.js

```diff
@@ -146,22 +168,66 @@ export default class Editor extends BaseClass {
         this.meta(value === this.field() ? undefined : value.replace(rxScriptStriper, ''));
     }
 
+    writeImage(value) {
+        var checkedValue = this.field() === value ? undefined : value.replace(rxScriptStriper, '');
+        this.meta({
+            src: checkedValue
+        });
+    }
+
     validateImage(params) {
-        var imageSrc = this.article.meta[params.src],
-            image = imageSrc(),
+        var imageSrc = this.article.meta[params.imageSource],
+            image = imageSrc().src,
             opts = params.options;
 
-
         if (image) {
             let {src, origin} = extractImageElements(image);
             return validateImageSrc(src, opts)
                 .then(img => {
-                    this.assignImageToSpreadElement(params, img, origin || src);
+                    if (this.type === 'text') {
+                        this.assignImageToSpreadElement(params, img, origin || src);
+                    } else if (this.type === 'text-image') {
+                        this.assignImageToObjectElement(params, img, origin || src);
+                    }
                 }, err => {
-                    this.assignImageToSpreadElement(params, err);
+                    if (this.type === 'text') {
+                        this.assignImageToSpreadElement(params, err);
+                    } else if (this.type === 'text-image') {
+                        this.assignImageToObjectElement(params, err);
+                    }
                 });
         } else {
-            this.assignImageToSpreadElement(params, null);
+
+            if (this.type === 'text') {
+                this.assignImageToSpreadElement(params, null);
+            } else if (this.type === 'text-image') {
+                this.assignImageToObjectElement(params, null);
+            }
+        }
+    }
+
+    assignImageToObjectElement(params, imgOrError, origin) {
+
+        if (!imgOrError || imgOrError instanceof Error) {
+
+            assign(this, [this.article.meta.imageSource], [{
```

Comment:
instead of assigning an object with empty values, it's better assigning undefined